### ISSUE TITLE
Testflight Error: ITC.apps.tf.build.test.info.missing.feedback 

### ISSUE THREAD

**Issue by m-ruhl** [fastlane/fastlane#8991]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Since fastlane 2.28.0 our internal testflight distribution fails.
The Feedback email is set in iTunesConnect.

```
00:12:48.422 Spaceship::Client::UnexpectedResponse: [!] {"code"=>"ITC.apps.tf.build.test.info.missing.feedback", "description"=>"Missing feedback email."}
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/spaceship/lib/spaceship/test_flight/client.rb:118:in `handle_response'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/spaceship/lib/spaceship/test_flight/client.rb:77:in `put_build'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/spaceship/lib/spaceship/test_flight/build.rb:154:in `save!'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/spaceship/lib/spaceship/test_flight/build.rb:161:in `update_build_information!'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/pilot/lib/pilot/build_manager.rb:51:in `distribute'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/pilot/lib/pilot/build_manager.rb:40:in `upload'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/actions/pilot.rb:15:in `run'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:252:in `block (2 levels) in execute_action'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:230:in `block in execute_action'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:226:in `chdir'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:226:in `execute_action'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
00:12:48.422   Fastfile:98:in `block (2 levels) in parsing_binding'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/lane.rb:33:in `call'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:45:in `chdir'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/runner.rb:45:in `execute'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/commander-fastlane-4.4.4/lib/commander/command.rb:178:in `call'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/commander-fastlane-4.4.4/lib/commander/command.rb:153:in `run'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/commander-fastlane-4.4.4/lib/commander/runner.rb:476:in `run_active_command'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:40:in `run!'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/commander-fastlane-4.4.4/lib/commander/delegates.rb:15:in `run!'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/fastlane/lib/fastlane/cli_tools_distributor.rb:66:in `take_off'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/gems/fastlane-2.28.3/bin/fastlane:20:in `<top (required)>'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/bin/fastlane:23:in `load'
00:12:48.422   /Users/ci/.rvm/gems/ruby-2.3.0/bin/fastlane:23:in `<top (required)>'
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                           |
| --------------------------- | ----------------------------------------------- |
| OS                          | 10.12.1                                         |
| Ruby                        | 2.3.0                                           |
| Bundler?                    | false                                           |
| Git                         | git version 2.11.0                              |
| Installation Source         | ~/.rvm/gems/ruby-2.3.0/bin/fastlane             |
| Host                        | Mac OS X 10.12.1 (16B2555)                      |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.3.0/lib                    |
| OpenSSL Version             | OpenSSL 1.0.2h  3 May 2016                      |
| Is contained                | false                                           |
| Is homebrew                 | false                                           |
| Is installed via Fabric.app | false                                           |
| Xcode Path                  | /Applications/Xcode-8.3.app/Contents/Developer/ |
| Xcode Version               | 8.3                                             |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     |             |   |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:
<details><summary>`./fastlane/Fastfile`</summary>

```ruby
platform :ios do

  ############################## PRE ##############################

  before_all do |lane, params|

    bundle_update
    install_cocoapods
  end

  ######################### PUBLIC LANES ##########################
  desc "• Run Unit- and UI-Tests"
  desc "• Generate Documentation"
  desc "• Submit the version to Testflight"
  desc "• Create git tag"
  lane :integration do
    ENV['SCAN_SKIP_BUILD'] = 'true'

    # verify git
    ensure_git_branch

    # build app
    match
    _build(
      gym: {
        include_symbols: false # workaround for: https://github.com/fastlane/fastlane/issues/4150
      }
    )

    # run test
    tests

    docu

    # submit to testflight
    pilot(
      changelog: changelog_from_git_commits(tag_match_pattern: BUILD_TAG_PATTERN)
    )

    # upload dsym to hockeyapp
    hockey(
      upload_dsym_only: true,
      release_type: "1",
      public_identifier: HOCKEY_APP_PUBLIC
    )

    # tag build
    add_git_tag(
      tag: "builds/#{lane_context[SharedValues::VERSION_NUMBER]}"
    )
    push_git_tags

    update_jira(state: "Integrated")

  end

end
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.28.3  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin               | Version | Update-Status |
| -------------------- | ------- | ------------- |
| fastlane-plugin-brew | 0.1.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| terminal-notifier         | 1.7.1        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.55.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| multi_json                | 1.12.1       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.12.5       |
| faraday_middleware        | 0.11.0.1     |
| json                      | 2.1.0        |
| io-console                | 0.4.5        |
| rubyzip                   | 1.2.1        |
| faraday                   | 0.12.1       |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| fastimage                 | 2.1.0        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| fastlane-plugin-brew      | 0.1.1        |
</details>

*generated on:* **2017-04-25**
</details>

**Comment by younatics** [comment]
+1

**Comment by tache** [comment]
See #8954 for full thread ... errors due to TestFlight changes ... you need to add the following to pilot ... but you will hit another issue further downstream 
```
beta_app_description: "My great app for iPad",
beta_app_feedback_email: "email@domain.com",
```

**Comment by mpirri** [comment]
@m-ruhl @younatics Can you update to the latest version of _fastlane_ and see if this is still happening for you?  You will probably need to add a `beta_app_feedback_email` if you do not have one, but I'm curious to know if that is required or not.

Thanks so much!!

**Comment by thasegaw** [comment]
I tested it  in fastlan-2.28.4 after I filled test information on testflight.

### success
```ruby
pilot

or

pilot(
  changelog: 'test test', 
  beta_app_feedback_email: 'test@example.com'
  )
```

### fail
#### set only `changelog`
```ruby
pilot(changelog: 'test test')
```
##### result 
pilot require `beta_app_feedback_email` parameter
```
{"code"=>"ITC.apps.tf.build.test.info.missing.feedback", "description"=>"Missing feedback email."}
```

#### set only `beta_app_feedback_email`
```ruby
pilot(beta_app_feedback_email: 'test@example.com')
```
##### result
pilot require `changelog` parameter.
```
{"code"=>"ITC.apps.tf.build.test.info.missing.whats.new", "description"=>"Missing what's new info."}
```
 
Is this a bug in `update_build_information`? 
These parameters are optional in `update_build_information`.
https://github.com/fastlane/fastlane/blob/50aa926104f1bd3d65156b96aa4614ede140f6fb/spaceship/lib/spaceship/tunes/tunes_client.rb#L816-L820

**Comment by Daredevilll** [comment]
I already added beta_app_description and beta_app_feedback_email parameters to my pilot.
Still I had this error :

`{"code"=>"ITC.apps.tf.build.missing.primary.test.info", "description"=>"Missing primary test info."}`

-> Adding changelog parameter to pilot resolved my issue  👍

`changelog: "Test Test",`

**Comment by mpirri** [comment]
@thasegaw Awesome observations. I believe you are correct - `beta_app_feedback_email` should only be required for external testing.  `changelog` as well.  I will do some testing to see if my observations match yours.

These errors are coming from the iTunesConnect backend so it means either:
- Apple changed the rules and requires things that we previously thought were optional
- We are calling an iTunesConnect API as part of our internal distribution that is only required for external distribution

I'll research further and report back!

**Comment by mpirri** [comment]
@thasegaw We just shipped _fastlane_ 2.28.5 with changes to internal testing - I'm not currently able to reproduce the behavior you observed above using this release. Can you try again and see if you are still able to reproduce the behavior when you call `pilot` with only `changelog` or `beta_app_feedback_email`?

Thanks!

**Comment by thasegaw** [comment]
@mpirri I observed following pattern in _fastlane_ 2.28.5. 
1. `pilot` with only `changelog`
2. `pilot` with only `beta_app_feedback_email`

The result is that same error has occurred in both of them.

**Comment by m-ruhl** [comment]
I can confirm @thasegaw observations.

**Comment by thasegaw** [comment]
It seems that this function change value to null when the parameter does not set.
https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/test_flight/build.rb#L151

```
$ fastlane spaceship                                                                                                                                 
Username: test@example.com
Logging into to iTunes Connect (test@example.com)...
Successfully logged in to iTunes Connect

Logging into the Developer Portal (test@example.com)...
Successfully logged in to the Developer Portal

---------------------------------------
| Welcome to the spaceship playground |
---------------------------------------

Enter docs to open up the documentation
Enter exit to exit the spaceship playground
Enter _ to access the return value of the last executed command

Just enter the commands and confirm with Enter
[1] pry(#<Spaceship::Playground>)> build = Spaceship::TestFlight::Build.find(app_id: '1111111111', build_id: '22222222')
=> <Spaceship::TestFlight::Build
    bundle_id="jp.co.test.bundle.id",
    internal_state="testflight.build.state.testing.active",
    external_state="testflight.build.state.submit.ready",
    id=19607356,
    train_version="1.0",
    build_version="110",
    beta_review_info=<Spaceship::TestFlight::BetaReviewInfo
        contact_first_name="Foo",
        contact_last_name="Bar",
        contact_phone="XXXXXXXXXXX",
        contact_email="test@example.com",
        demo_account_name=nil,
        demo_account_password=nil,
        demo_account_required=true>,
    export_compliance=<Spaceship::TestFlight::ExportCompliance
        uses_encryption=nil,
        encryption_updated=nil>,
    test_info=<Spaceship::TestFlight::TestInfo
        description=nil,
        feedback_email="default@example.com",
        whats_new=nil>,
    install_count=0,
    invite_count=0,
    crash_count=0,
    did_notify=false,
    upload_date=2017-04-28 03:05:28 +0000>
[2] pry(#<Spaceship::Playground>)> build.to_json
=> "{\"id\":12345678, **************
*
*
"testInfo\":[{\"locale\":\"ja\",\"primaryLocale\":true,\"description\":null,\"feedbackEmail\":\"default@example.com\",\"marketingUrl\":null,\"privacyPolicyUrl\":null,\"privacyPolicy\":null,\"whatsNew\":null}]
*
*
}"
[3] pry(#<Spaceship::Playground>)> build.update_build_information!(description: 'testtest')
Spaceship::Client::UnexpectedResponse: {"code"=>"ITC.apps.tf.build.test.info.missing.whats.new", "description"=>"Missing what's new info."}
from /Users/thasegaw/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.28.5/spaceship/lib/spaceship/test_flight/client.rb:133:in `handle_response'
[4] pry(#<Spaceship::Playground>)> build.to_json
=> "{\"id\":12345678, **************
*
*
"testInfo\":[{\"locale\":\"ja\",\"primaryLocale\":true,\"description\":\"testtest\",\"feedbackEmail\":null,\"marketingUrl\":null,\"privacyPolicyUrl\":null,\"privacyPolicy\":null,\"whatsNew\":null}]
*
*
}"
```

**Comment by mpirri** [comment]
@thasegaw We just shipped _fastlane_ 2.28.6 with a fix for the bug in the `Build` model object you describe above.  Can you give it a try to see if it works better for you? Thanks again!

**Comment by thasegaw** [comment]
@mpirri Thank you for your PR! I confirmed this bug was fixed in _fastlane_ 2.28.6. 👍 
```
[2] pry(#<Spaceship::Playground>)> build.update_build_information!(changelog: 'hogehoge')
ArgumentError: unknown keyword: changelog
from /Users/thasegaw/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/fastlane-2.28.6/spaceship/lib/spaceship/test_flight/build.rb:170:in `update_build_information!'
[3] pry(#<Spaceship::Playground>)> build.to_json

*
*
"testInfo\":[{\"locale\":\"ja\",\"primaryLocale\":true,\"description\":null,\"feedbackEmail\":\"default@example.com\",\"marketingUrl\":null,\"privacyPolicyUrl\":null,\"privacyPolicy\":null,\"whatsNew\":null}]
*
*
[4] pry(#<Spaceship::Playground>)> build.update_build_information!(whats_new: 'testtest')
=> {"id"=>19607356,
*
*
  [{"locale"=>"ja",
    "primaryLocale"=>true,
    "description"=>nil,
    "feedbackEmail"=>"default@example.com",
    "marketingUrl"=>nil,
    "privacyPolicyUrl"=>nil,
    "privacyPolicy"=>nil,
    "whatsNew"=>"testtest"}],
*
*
```

**Comment by mpirri** [comment]
Great news! I'm going to close this issue now, but please let us know if you encounter any other difficulty with this.
